02/userinfamous/game.py

```diff
@@ -0,0 +1,97 @@
+#!python3
+# Code Challenge 02 - Word Values Part II - a simple game
+# http://pybit.es/codechallenge02.html
+import re
+from random import choice
+from data import DICTIONARY, LETTER_SCORES, POUCH
+from collections import defaultdict
+
+NUM_LETTERS = 7
+
+
+# re-use from challenge 01
+def calc_word_value(word: str) -> int:
+    """Calc a given word value based on Scrabble LETTER_SCORES mapping"""
+    return sum(LETTER_SCORES.get(char.upper(), 0) for char in word)
+
+
+# re-use from challenge 01
+def max_word_value(words: set) -> str:
+    """Calc the max value of a collection of words"""
+    return max(words, key=calc_word_value)
+
+# check if user input word can be made from the letters
+def input_check(word: str, pouch: list) -> bool:
+    matched = 0
+    for letter in word:
+        if letter.upper() in pouch:
+            matched += 1
+    return True if matched == len(word) else False
+
+# check if it matches the dictionary
+def dictionary_check(word: str, dictionary: set) -> None:
+    print('\n+ User enterred a valid input. \n+ Checking score...')
+    if re.search(word, ''.join(dictionary), re.IGNORECASE):
+        score = calc_word_value(word)
+        print('+ Word score is ', score)
+        return score
+    else:
+        print('- Word is not found')
+        return
+
+# function to reduce and find the best word
+def best_word(pouch: list, dictionary: set) -> int:
```

Comment:
there are parts that can be moved out of this function it seems to make it easier / more reusable